Lesson17/main.js

```diff
@@ -0,0 +1,120 @@
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const addErrorMessage = (error, parent) => {
+  const errorMessage = createElementWithClassName("p", "error-message");
+  errorMessage.textContent = error;
+  parent.appendChild(errorMessage);
+}
+
+const fetchErrorHandling = async (response) => {
+  if (response.ok) {
+    return await response.json();
+  } else {
+    throw new Error("サーバーエラーが発生しました");
+  }
+}
+
+const getJsonOrError = async (url) => {
+  const response = await fetch(url);
+  const json = await fetchErrorHandling(response);
+  return json;
+}
+
+const getSlideImageData = async () => {
+  return new Promise((resolve) => {
+    setTimeout(() => resolve(getJsonOrError("https://myjson.dit.upm.es/api/bins/gzcn")), 3000);
+  })
+}
+
+const tryGetSlideImageData = async () => {
+  addLoading(slideWrapper);
+  try {
+    return await getSlideImageData();
+  } catch (e) {
+    addErrorMessage(e, slideWrapper);
+  } finally {
+    removeLoading();
+  }
+}
+
+const init = async () => {
+  const result = await tryGetSlideImageData();
+  if (result) {
+    createSlideItem(result);
+    createArrowButtons();
+  }
+}
+
+const createSlideItem = (imageData) => {
+  for (let i = 0; i < imageData.length; i++) {
+    const slideItem = createElementWithClassName("li", "slide-item");
+    const slideImage = createElementWithClassName("img", "slide-img");
+    slideImage.src = imageData[i].image;
+    // 初期設定として 最初の要素にis-displayingクラスを付与
+    if (i === 0) slideItem.classList.add("is-displaying");
+    slideItem.appendChild(slideImage);
+    slideList.appendChild(slideItem);
+  }
+}
+
+const createArrowButtons = () => {
+  const arrowBtnWrapper = createElementWithClassName("div", "arrow-btn__wrapper");
+  const prevButton = createElementWithClassName("button", "arrow-btn --prev");
+  const nextButton = createElementWithClassName("button", "arrow-btn --next");
+  prevButton.appendChild(document.createElement("span"));
+  nextButton.appendChild(document.createElement("span"));
+  prevButton.id = "js-arrow-left-btn";
+  nextButton.id = "js-arrow-right-btn";
+  arrowBtnWrapper.appendChild(prevButton);
+  arrowBtnWrapper.appendChild(nextButton);
+  slideWrapper.appendChild(arrowBtnWrapper)
+  setClickEventInArrowButton();
+}
+
+const setClickEventInArrowButton = () => {
+  const arrowButtons = document.querySelectorAll(".arrow-btn");
+  arrowButtons.forEach(button => {
+    button.addEventListener("click", (e) => {
+      const displayingItem = document.querySelector(".is-displaying");
+      if (e.currentTarget.id === "js-arrow-right-btn") {
+        clickNextButton(displayingItem);
+      } else {
+        clickPrevButton(displayingItem);
+      }
+    });
+  });
+}
+
+const clickPrevButton = (target) => {
+  const firstSlideItem = slideList.firstElementChild;
+  if (target !== firstSlideItem) {
+    target.classList.remove("is-displaying");
+    target.previousElementSibling.classList.add("is-displaying");
+  }
+}
```

Comment:
> Is the click event to the arrow complicated?
I was wondering if I should control the if statement by index number or use firstElementChild or lastElementChild as in this implementation.

I'm glad to know the new way to implement this part. 
I think it's no problem, this is also one of the way to implement. But I would share with you what I used here when I implemented this button event.
I used `previousElementSibling` and `nextElementSibling` insted of `firstElementChild` and index. I hope this give you new idea.